wrappers/rust/icicle-core/src/ntt/mod.rs

```diff
@@ -72,6 +72,18 @@ impl<'a, S: FieldImpl> NTTConfig<'a, S> {
             is_async: false,
         }
     }
+    pub fn default_config_for_device(device_id: usize) -> Self {
```

Comment:
Maybe let's leave one function? Either `default_config_for_device` deleting the existing `default_config` or without this function, just setting the device to 0 by default

Reply:
removing old is a breaking change and we need new function anyway

Reply:
Yea I don't like the idea of removing the old function either. But why letting the user explicitly set device here? Especially considering that we want there to be one device per one CPU thread so no explicit device changes besides setting it for the duration of a thread right? I think you mentioned offline that that was the idea. And proper device id can always be accessed by [`cudaGetDevice`](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g80861db2ce7c29b6e8055af8ae01bc78) too...
But if you insist on this scheme - let's be consistent. You created an extra config creation function that accepts a `device_id` for NTT but not for MSM, I think it should be either both of them or neither.